torch/nn/functional.py

```diff
@@ -1960,11 +1960,7 @@ def pairwise_distance(x1, x2, p=2, eps=1e-6):
         >>> output = F.pairwise_distance(input1, input2, p=2)
         >>> output.backward()
     """
-    assert x1.size() == x2.size(), "Input sizes must be equal."
-    assert x1.dim() == 2, "Input must be a 2D matrix."
-    diff = torch.abs(x1 - x2)
-    out = torch.pow(diff + eps, p).sum(dim=1, keepdim=True)
```

Comment:
I'm a little worried because it looks like we're changing the behavior of `pairwise_distance` to return a tensor of size `(N,)` instead of size `(N, 1)`.

I think keeping the old behavior (return tensor of size `(N, 1)`) might be better because it represents batched distances. On the other hand, this makes it inconsistent with our loss functions (but I guess `pairwise_distance` isn't a loss, and you can keep the current behavior of `triplet_margin_loss`. What do you think?

Reply:
The reason I did this was because legacy seems to return (N) size. https://github.com/pytorch/pytorch/blob/master/torch/legacy/nn/PairwiseDistance.py#L30

I'm not sure why the change was made, but I think we should make them the same. (By changing legacy behavior to (N, 1) maybe?)

Reply:
If any users use `pairwise_distance` right now (the non-legacy version) we'll be breaking their code. Not sure if that's a good thing or a bad thing. We should definitely keep the legacy code the same, it's there for interfacing with users of lua torch.

Reply:
Pairwise distance takes the distance for each element to each element in another set (NxM) (http://scikit-learn.org/stable/modules/generated/sklearn.metrics.pairwise.pairwise_distances.html).
This function calculates one distance between two vectors (N).

Since it's a relatively fresh PR I'm asking here: @li-roy can we have some clarification?